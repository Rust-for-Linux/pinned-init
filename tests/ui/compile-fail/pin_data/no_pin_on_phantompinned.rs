use pin_init::*;
use std::marker::{self, PhantomPinned};

#[pin_data]
struct Foo {
    pin1: PhantomPinned,
    pin2: marker::PhantomPinned,
    pin3: core::marker::PhantomPinned,
    pin4: ::core::marker::PhantomPinned,
}

fn main() {}
