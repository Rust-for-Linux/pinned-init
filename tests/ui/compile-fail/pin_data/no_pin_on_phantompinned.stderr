error: The field `pin1` of type `PhantomPinned` only has an effect, if it has the `#[pin]` attribute
 --> tests/ui/compile-fail/pin_data/no_pin_on_phantompinned.rs:4:1
  |
4 | #[pin_data]
  | ^^^^^^^^^^^
  |
  = note: this error originates in the attribute macro `pin_data` (in Nightly builds, run with -Z macro-backtrace for more info)

error: The field `pin2` of type `PhantomPinned` only has an effect, if it has the `#[pin]` attribute
 --> tests/ui/compile-fail/pin_data/no_pin_on_phantompinned.rs:4:1
  |
4 | #[pin_data]
  | ^^^^^^^^^^^
  |
  = note: this error originates in the attribute macro `pin_data` (in Nightly builds, run with -Z macro-backtrace for more info)

error: The field `pin3` of type `PhantomPinned` only has an effect, if it has the `#[pin]` attribute
 --> tests/ui/compile-fail/pin_data/no_pin_on_phantompinned.rs:4:1
  |
4 | #[pin_data]
  | ^^^^^^^^^^^
  |
  = note: this error originates in the attribute macro `pin_data` (in Nightly builds, run with -Z macro-backtrace for more info)

error: The field `pin4` of type `PhantomPinned` only has an effect, if it has the `#[pin]` attribute
 --> tests/ui/compile-fail/pin_data/no_pin_on_phantompinned.rs:4:1
  |
4 | #[pin_data]
  | ^^^^^^^^^^^
  |
  = note: this error originates in the attribute macro `pin_data` (in Nightly builds, run with -Z macro-backtrace for more info)
